### PR TITLE
Update to PHP7.3+ and PHPStan 1.0+

### PR DESCRIPTION
**Pull request by dereuromark**
Also 
- CodeSniffer upgrade
- PHP 8.1 in test matrix
- Removal of very outdated Symf v3

TODO in follow up: Adding Symf v6 compatibility

Note:
ControlStructures.AssignmentInCondition.AssignmentInCondition is silenced for now as this creates too much manual work
This can and should be a follow up PR.

The failing remaining CS issue is fixed in code sniffer and will be green once we merge.

### CODE REVIEW

**Comment by dereuromark** [comment]
anyone else being able to review?

File: src/Propel/Runtime/ActiveQuery/ModelCriteria.php [hunk 1]
```diff
@@ -1920,7 +1935,7 @@ public function doUpdate($updateValues, ConnectionInterface $con, $forceIndividu
     {
         if ($forceIndividualSaves) {
             // Update rows one by one
-            $objects = $this->setFormatter(ModelCriteria::FORMAT_OBJECT)->find($con);
+            $objects = $this->setFormatter(self::FORMAT_OBJECT)->find($con);
```
Comment (gechetspr) [inline src/Propel/Runtime/ActiveQuery/ModelCriteria.php:1938]: mb we can use static here?

Comment (gechetspr) [reply src/Propel/Runtime/ActiveQuery/ModelCriteria.php:1938]: But I guess static can change behavior here

File: src/Propel/Runtime/Map/TableMap.php [hunk 2]
```diff
@@ -391,35 +400,6 @@ public function getPrimaryKeyMethodInfo()
...
 
-    /**
-     * Helper method which returns the primary key contained
-     * in the given Criteria object.
-     *
-     * @param \Propel\Runtime\ActiveQuery\Criteria $criteria A Criteria.
-     *
-     * @return \Propel\Runtime\Map\ColumnMap If the Criteria object contains a primary key, or null if it doesn't.
-     */
-    private static function getPrimaryKey(Criteria $criteria)
```
Comment (gechetspr) [inline src/Propel/Runtime/Map/TableMap.php:402]: this is not used?

**Review by gechetspr — COMMENTED** [review]
Looks good, but I have a few questions

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -5,32 +5,30 @@ on:
...
-    - cron: "0 0 * * *"
 
 jobs:
     testsuite:
         runs-on: ubuntu-18.04
         strategy:
             fail-fast: false
             matrix:
-                php-version: [ '7.2', '7.4', '8.0' ]
+                php-version: [ '7.3', '7.4', '8.1' ]
```
[the remark this answers is not in the document]
Comment (dereuromark) [reply .github/workflows/ci.yml:15]: Maybe a bit soon?

File: src/Propel/Generator/Behavior/AggregateMultipleColumns/AggregateMultipleColumnsBehavior.php [hunk 4]
```diff
@@ -60,7 +83,7 @@ class AggregateMultipleColumnsBehavior extends Behavior
      */
     public static function resetInsertedAggregationNames(): void
     {
-        static::$insertedAggregationNames = [];
+        self::$insertedAggregationNames = [];
```
[the remark this answers is not in the document]
Comment (dereuromark) [reply src/Propel/Generator/Behavior/AggregateMultipleColumns/AggregateMultipleColumnsBehavior.php:86]: Yeah

File: .github/workflows/ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dereuromark) [reply .github/workflows/ci.yml:15]: We can make a follow up PR to add 7.4+, maybe?
What do others think? Shall we do that?

**Pull request merged**